### PR TITLE
Jup-13 Join Org Functionality

### PR DESCRIPTION
**Pull request by KennethImphean**
Created a Join Button component for users to join clubs along with API procedures to update User_Meta_Data_To_Clubs table in database when users join. The Org Headers and Org Directory components were updated to incorporate the button. When users join a club, the club will then be removed from the Org Directory so that new orgs can be discovered by users.

### CODE REVIEW

File: src/components/JoinButton.tsx [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+      mutation.mutate({clubid});
+      setDisabled(!isDisabled);
+    }
+    if(!session){
+        return (
+          <button 
+            className={isHeader ? "rounded-full px-8 py-4 text-xs font-extrabold text-white disabled:bg-slate-700":
+            "mr-2 rounded-2xl px-4 py-2 text-xs font-extrabold text-white border-solid disabled:bg-slate-700"
+            }
+            disabled={true}
```
Comment (RubenOlano) [inline src/components/JoinButton.tsx:34]: Can just rewrite this as `disabled`

File: src/components/JoinButton.tsx [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+    const [isDisabled, setDisabled] = useState(isJoined??false);
+    const handleJoin = () => { 
+      mutation.mutate({clubid});
+      setDisabled(!isDisabled);
+    }
+    if(!session){
+        return (
+          <button 
+            className={isHeader ? "rounded-full px-8 py-4 text-xs font-extrabold text-white disabled:bg-slate-700":
+            "mr-2 rounded-2xl px-4 py-2 text-xs font-extrabold text-white border-solid disabled:bg-slate-700"
```
Comment (RubenOlano) [inline src/components/JoinButton.tsx:32]: You can rewrite this as 
```
className={`text-xs font-extrabold text-white disabled:bg-slate-700 ${isHeader ? "rounded-full px-8 py-4" : "mr-2 rounded-2xl px-4 py-2 border-solid"
```
I would also add a mouse disabled state, something like `cursor-not-allowed`

File: src/components/JoinButton.tsx [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+          >
+            Join
+          </button> 
+        );
+      }
+    return (
+      <button 
+        onClick={handleJoin}
+        className={isHeader ? "bg-blue-primary disabled:bg-blue-700 rounded-full  px-8 py-4 text-xs font-extrabold text-white transition-colors hover:bg-blue-700":
+        "bg-blue-primary disabled:bg-blue-700 mr-2 rounded-2xl px-4 py-2 text-xs font-extrabold text-white transition-colors hover:bg-blue-700 "}
```
Comment (RubenOlano) [inline src/components/JoinButton.tsx:44]: Try to combine this one like the comment I left above

File: src/components/OrgDirectoryGrid.tsx [hunk 4]
```diff
@@ -1,8 +1,15 @@
...
 
 const OrgDirectoryGrid = async ({ tag }: { tag?: string }) => {
-  const clubs = await api.club.all.query({ tag });
+  const session = await getServerAuthSession();
+  let clubs;
+  if(session){
+    clubs = await api.club.allButJoined.query({ tag });
+  } else{
+    clubs = await api.club.all.query({ tag });
+  }
```
Comment (RubenOlano) [inline src/components/OrgDirectoryGrid.tsx:12]: Try moving this logic into the server. COmbine the `all` and `allButJoined` into one query

**Review by RubenOlano — CHANGES_REQUESTED** [review]
Great looking PR, some changes for the styling stuff but also you can combine the two procedures to make it a little more clean

**Comment by KennethImphean** [comment]
The changes to styling and combining of the two procedures has been made!

**Review by RubenOlano — APPROVED** [review]
👍🏽

**Pull request merged**